test/ng/documentSpec.js

```diff
@@ -27,3 +27,29 @@ describe('$document', function() {
     });
   });
 });
+
+
+describe('$$isDocumentHidden', function() {
```

Comment:
I don't think you need all that. I think all you need it mocking `document` wrt the `hidden` attribute (e.g. making it a getter/setter). The events are dispatched through `$document`, so you can trigger them using `triggerHandler('visibilitychange')`.

But this is aff the top of my head, so I might be missing something. If it's not too complicated I'd say let's add some.
If it's too complicated, it's fine leaving it out.